### ISSUE TITLE
RUN demo has error

### ISSUE THREAD

**Issue by vVVtreasure** [peiyunh/tiny#31]
Loading pretrained detector model...
Processing data/demo/selfie.jpg at scale 0.031250.
Processing data/demo/selfie.jpg at scale 0.062500.
Processing data/demo/selfie.jpg at scale 0.125000.
Processing data/demo/selfie.jpg at scale 0.250000.
Processing data/demo/selfie.jpg at scale 0.500000.
Processing data/demo/selfie.jpg at scale 1.000000.
Processing data/demo/selfie.jpg at scale 1.414214.
Processing data/demo/selfie.jpg at scale 2.000000.
error use  vl_nnbnorm
**Out of memory on device. To view more detail about available memory on the GPU, use 'gpuDevice()'. If the problem persists,**
reset the GPU by calling 'gpuDevice(1)'.

error use dagnn.BatchNorm/forward (line 15)
        outputs{1} = vl_nnbnorm(inputs{1}, params{1}, params{2}, ...

error use dagnn.Layer/forwardAdvanced (line 85)
      outputs = obj.forward(inputs, {net.params(par).value}) ;

error use dagnn.DagNN/eval (line 91)
  obj.layers(l).block.forwardAdvanced(obj.layers(l)) ;

error use tiny_face_detector (line 130)
  net.eval(inputs);

my gpu is 4g memory
 gpuDevice(1)

ans = 

  CUDADevice:

                      Name: 'GeForce GTX 1050 Ti'
                     Index: 1
         ComputeCapability: '6.1'
            SupportsDouble: 1
             DriverVersion: 8
            ToolkitVersion: 5.5000
        MaxThreadsPerBlock: 1024
          MaxShmemPerBlock: 49152
        MaxThreadBlockSize: [1024 1024 64]
               MaxGridSize: [2.1475e+09 65535 65535]
                 SIMDWidth: 32
               TotalMemory: 4.2950e+09
                FreeMemory: 3.5271e+09
       MultiprocessorCount: 6
              ClockRateKHz: 1392000
               ComputeMode: 'Default'
      GPUOverlapsTransfers: 1
    KernelExecutionTimeout: 1
          CanMapHostMemory: 1
           DeviceSupported: 1
            DeviceSelected: 1

how can I deal with this error??

**Comment by peiyunh** [comment]
This seems an out-of-memory issue, try changing this line (https://github.com/peiyunh/tiny/blob/master/tiny_face_detector.m#L48) from 

`MAX_INPUT_DIM = 5000;`

to 

`MAX_INPUT_DIM = 2000;`

or even smaller value.

**Comment by Mehreenghaafoor** [comment]
For what purpose we used MAX_INPUT_DIM= 5000;
